### PR TITLE
Database clean-up

### PR DESCRIPTION
**Pull request by D4nte**
Small clean-ups I spotted when reviewing #85.

### CODE REVIEW

**Review by da-kami — APPROVED** [review]

**Review by rishflab — APPROVED** [review]

File: swap/tests/happy_path.rs [hunk 1]
```diff
@@ -94,8 +92,8 @@ async fn happy_path() {
         Uuid::new_v4(),
     );
 
-    let _bob_swarm_fut = tokio::spawn(async move { bob_event_loop.run().await });
-
+    tokio::spawn(async move { alice_event_loop.run().await });
+    tokio::spawn(async move { bob_event_loop.run().await });
```
Comment (da-kami) [inline swap/tests/happy_path.rs:96]: Could the stackoverflow problem be related to removing the futures being kept in scope with `_` variables?
I don't think it should, but that is the only notable change I can see that could cause this.

Comment (D4nte) [reply swap/tests/happy_path.rs:96]: The  `JoinHandle` returned by `tokio::spawn` is only meant to be used to await on the stack if one wishes to do so.

File: xmr-btc/src/bob.rs [hunk 2]
```diff
@@ -373,6 +371,8 @@ impl State0 {
         let s_b = cross_curve_dleq::Scalar::random(rng);
         let v_b = monero::PrivateViewKey::new_random(rng);
 
+        let dleq_proof_s_b = cross_curve_dleq::Proof::new(rng, &s_b);
```
Comment (D4nte) [inline xmr-btc/src/bob.rs:374]: @luckysori what do you think? Do you see any downsize in doing so?

Comment (luckysori) [reply xmr-btc/src/bob.rs:374]: Nope, seems okay!

By the way, I would still recommend replacing this proof with Lloyd's version of it in `secp256kfun`. It's faster, smaller and more likely to be correct. Just a reminder.

Comment (D4nte) [reply xmr-btc/src/bob.rs:374]: Could you open an issue to describe in a bit more details please?

Comment (luckysori) [reply xmr-btc/src/bob.rs:374]: Is it okay if instead I open a simple issue and I do it quickly once you think the codebase is a bit more stable?

I can write a detailed explanation of what to do, but it feels like a waste of time.

Comment (luckysori) [reply xmr-btc/src/bob.rs:374]: https://github.com/comit-network/xmr-btc-swap/issues/90

Comment (D4nte) [reply xmr-btc/src/bob.rs:374]: > Is it okay if instead I open a simple issue and I do it quickly once you think the codebase is a bit more stable?
> 
> I can write a detailed explanation of what to do, but it feels like a waste of time.

Sounds great, thanks!

File: swap/src/alice/swap.rs [hunk 3]
```diff
@@ -502,22 +502,22 @@ impl Swap {
                         let refund_tx_seen =
                             bitcoin_wallet_clone.watch_for_raw_transaction(tx_refund.txid());
 
-                        pin_mut!(punish_tx_finalised);
-                        pin_mut!(refund_tx_seen);
-
-                        state = match select(punish_tx_finalised, refund_tx_seen).await {
-                            Either::Left(_) => AliceState::Punished,
-                            Either::Right((published_refund_tx, _)) => {
+                        select! {
```
Comment (da-kami) [inline swap/src/alice/swap.rs:505]: You can actually return the state from `select!` and assign it to the variable outside. 
Did you implement it with state being set inside on purpose?

**Comment by da-kami** [comment]
Note that there is another `pin_mut` in `steps.rs` that can also be replaced with `select!` - I tried that, but am still running into the `stackoverlfow` issue.

File: swap/src/alice/swap.rs (hunk 3 above)
Comment (D4nte) [reply swap/src/alice/swap.rs:505]: I was convinced of the opposite. Thanks!

**Comment by D4nte** [comment]
> Note that there is another `pin_mut` in `steps.rs` that can also be replaced with `select!` - I tried that, but am still running into the `stackoverlfow` issue.

I intend to check type sizes using `-Zprint-type-sizes`

**Comment by D4nte** [comment]
Close for now due to test failure

**Pull request closed without merging**